src/components/modes/teacher/TeacherView.js

```diff
@@ -161,10 +102,33 @@ export class TeacherView extends Component {
     dispatchGetUsers();
   }
 
-  handleChangeStudent = value => {
-    this.setState({
-      selectedStudent: value,
-    });
+  componentDidUpdate({ minDate: oldMinDate, maxDate: oldMaxDate }) {
+    const { minDate, maxDate } = this.props;
+    if (minDate !== oldMinDate) {
+      // eslint-disable-next-line react/no-did-update-set-state
+      this.setState({
+        fromDate: minDate,
+      });
+    }
+    if (maxDate !== oldMaxDate) {
+      // eslint-disable-next-line react/no-did-update-set-state
+      this.setState({
+        toDate: maxDate,
+      });
+    }
+  }
+
+  handleActionsDateRange = () => {
+    const { fromDate, toDate } = this.state;
+    const { actions } = this.props;
+    const rangedMoment = extendMoment(moment);
+    const start = rangedMoment.utc(fromDate);
+    const end = rangedMoment.utc(toDate);
+    const range = rangedMoment.range(start, end);
+    return Array.from(range.by('days')).map(day => ({
+      x: day.format('DD/MM/YYYY'),
+      y: actions.find(({ date }) => date.isSame(day, 'days'))?.count ?? 0,
```

Comment:
Are the `?.` and `??` operators being properly compiled when building the app?

Reply:
Yes, we should be fine. Babel `7.8.0` supports the two features. Anyway, I will double-check tomorrow and get back to you.

Reply:
👍